database/migrations/import_rmg_models.py

```diff
@@ -163,12 +165,31 @@ def create_and_save_species(kinetic_model, name, rmg_molecule, inchi="", **model
         raise
 
 
+def get_reaction_from_stoich_set(stoichiometry_data, **models):
+    filter_candidates = []
+    for stoich, species in stoichiometry_data:
+        filter_candidates = models["Reaction"].filter(
+            stoichiometry__stoichiometry=stoich, stoichiometry__species=species
```

Comment:
does this restrict the filter each time you iterate? I don't see how filter_candidates is preserved from one iteration to the next.

Reply:
Would be a good thing to unit test, in any case

Reply:
Yeah I will add a test for this. `Reaction.objects.filter()` (actually a typo in the source, oops) will give you a filtered django `QuerySet` that will have all `Reaction`s with that specified species and stoichiometric coefficient. I just filter down the set for every coefficient-species pair which should theoretically filter down to a single `Reaction` if the validation I wrote below works.